Firestore/Source/Local/FSTLevelDBKey.mm

```diff
@@ -255,10 +258,10 @@ BOOL ReadLabeledStringMatching(Slice *contents,
  * For each segment in the given resource path writes an FSTComponentLabelPathSegment component
  * label and a string containing the path segment.
  */
-void WriteResourcePath(std::string *dest, FSTResourcePath *path) {
-  for (int i = 0; i < path.length; i++) {
+void WriteResourcePath(std::string *dest, const ResourcePath &path) {
+  for (int i = 0; i < path.size(); i++) {
```

Comment:
This is fine.

You could optionally make ResourcePath into something that you could iterate over:
http://go/soverflow/8164567

(If you want to do this, it would probably be better in a followup, rather than this PR. Though you could consider a TODO here to remind you. (Again, this is all optional.))

Reply:
Done. Actually `ResourcePath` already supports that.

Reply:
BasePath already defines begin() and end() so you should be able to do this already.